### PR TITLE
Integrate builtin logic to `VirtualMachine.ComputeOperands`

### PR DESCRIPTION
**Pull request by fmoletta**

### CODE REVIEW

File: pkg/vm/vm_core.go [hunk 1]
```diff
@@ -325,13 +320,63 @@ func (vm *VirtualMachine) ComputeOperands(instruction Instruction) (Operands, er
...
+		op0, deduced_res, err = vm.DeduceOp0(instruction, dst, op1)
+		if err != nil {
+			return *memory.NewMaybeRelocatableFelt(lambdaworks.FeltZero()), nil, err
+		}
+	}
+	if op0 != nil {
+		vm.Segments.Memory.Insert(op0_addr, op0)
+	} else {
+		return *memory.NewMaybeRelocatableFelt(lambdaworks.FeltZero()), nil, errors.New("Failed to compute or deduce op0")
+	}
```
Comment (gabrielbosio) [inline pkg/vm/vm_core.go:351]: I don't understand these conditionals, the `else` from line 347 is the same as the condition from line 339, right?

Comment (fmoletta) [reply pkg/vm/vm_core.go:351]: line 340 updates the value of `op0` so the check on line 345 is checking that new value

Comment (fmoletta) [reply pkg/vm/vm_core.go:351]: The logic would be as follows : 
op0 is deduced (DeducedOp0) now op0 can be some or none
if op0 is none (line 339) deduce it via builtins (DeduceMemoryCell) now op0 can be some or none
Then is op0 is some (line 345), it was deduced by any of the two methods, so we need to insert it into memory
If op0 is still none, an error is returned

Comment (gabrielbosio) [reply pkg/vm/vm_core.go:351]: Makes sense

File: pkg/vm/vm_test.go [hunk 2]
```diff
@@ -507,6 +507,15 @@ func TestComputeOperandsAddAp(t *testing.T) {
 	}
 }
 
+func DeduceMemoryCellNoBuiltins(t *testing.T) {
```
Comment (gabrielbosio) [inline pkg/vm/vm_test.go]:
```suggestion
func TestDeduceMemoryCellNoBuiltins(t *testing.T) {
```

**Review by gabrielbosio — APPROVED** [review]
Looks good, I left a small suggestion

**Review by pablodeymo — APPROVED** [review]

**Pull request merged**